File: Tutorial4-6SubComponentBuildersMultipleModulesAndActvities/src/main/java/com/example/tutorial4_6subcomponentbuildersmultiplemodulesandactvities/di/ToastMakerModule.java
package com.example.tutorial4_6subcomponentbuildersmultiplemodulesandactvities.di;


import android.content.Context;

import com.example.tutorial4_6subcomponentbuildersmultiplemodulesandactvities.model.ToastMaker;

import dagger.Module;
import dagger.Provides;

/**
 * This is SubComponent Module of {@link ToastAndPreferencesSubComponent} in this example
 * A module can be used in multiple components
 *
 * Sub component and module CAN NOT have different scopes.
 * IMPORTANT: Both can be un- scoped but cannot have with different scopes
 */


@Module
public class ToastMakerModule {

    @FragmentScope
    @Provides
    ToastMaker provideToastMaker(Context context) {
        return new ToastMaker(context);
    }
}

File: Tutorial6-3SubComponentFactory/src/main/java/com/smarttoolfactory/tutorial6_3subcomponentfactory/di/ToastMakerModule.kt
package com.smarttoolfactory.tutorial6_3subcomponentfactory.di

import android.content.Context
import com.example.tutorial4_6subcomponentbuilders_multiplemodulesandactvities.di.scope.FragmentScope
import com.smarttoolfactory.tutorial6_3subcomponentfactory.model.ToastMaker
import dagger.Module
import dagger.Provides

/**
 * This is SubComponent Module of [ToastAndPreferencesSubComponent] in this example
 * A module can be used in multiple components
 *
 * Sub component and module CAN NOT have different scopes.
 * IMPORTANT: Both can be un- scoped but cannot have with different scopes
 */
@Module
class ToastMakerModule {
    @FragmentScope
    @Provides
    fun provideToastMaker(context: Context): ToastMaker {
        return ToastMaker(context)
    }
}